# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/clang/hotfix-tailcalls.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- clang/lib/AST/ByteCode/Interp.cpp.vanilla	2026-10-12 11:58:14.693843950 +0200
+++ clang/lib/AST/ByteCode/Interp.cpp	2026-10-12 11:59:32.314227637 +0200
@@ -41,7 +41,7 @@
 // to handle the way we use tailcalls.
 // PPC can't tail-call external calls, which is a problem for InterpNext.
 #if defined(_MSC_VER) || defined(__powerpc__) || !defined(MUSTTAIL) ||         \
-    defined(__i386__) || defined(__sparc__)
+    defined(__i386__) || defined(__sparc__) || defined(__alpha__)
 #undef MUSTTAIL
 #define MUSTTAIL
 #define USE_TAILCALLS 0
